include/ginkgo/core/base/vector_cache.hpp

```diff
@@ -0,0 +1,114 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_BASE_VECTOR_CACHE_HPP_
+#define GKO_PUBLIC_CORE_BASE_VECTOR_CACHE_HPP_
+
+
+#include <memory>
+
+#include <ginkgo/config.hpp>
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/base/mpi.hpp>
+#include <ginkgo/core/distributed/vector.hpp>
+
+
+#if GINKGO_BUILD_MPI
+
+
+namespace gko {
```

Comment:
I think it might make sense to put this into the `experimental` and `distributed` namespaces.

Reply:
I move it into `gko::experimental::distributed::detail`